File: package/main/src/Validate/isValueNaN.ts
/**
 * Determines if a value is NaN
 * @param value - Value to check
 * @param loose - If true, includes string values in NaN check (default: false)
 * @returns boolean - True if the value is NaN, false otherwise
 * @example
 * isValueNaN(1); // false
 * isValueNaN("NaN"); // false
 * isValueNaN("NaN", true); // true
 * isValueNaN(parseInt("not a number")); // true
 */
export const isValueNaN = (value: unknown, loose = false): boolean => {
  // biome-ignore lint/suspicious/noGlobalIsNan: <explanation>
  return loose ? isNaN(value as number) : Number.isNaN(value);
};

File: package/umt_python/src/validate/is_value_nan.py
import math


def is_value_nan(value: object, loose: bool = False) -> bool:
    """
    Determines if a value is NaN.

    Args:
        value: Value to check.
        loose: If True, includes string values in NaN check (default: False).

    Returns:
        True if the value is NaN, False otherwise.

    Example:
        >>> is_value_nan(1)
        False
        >>> is_value_nan("NaN")
        False
        >>> is_value_nan("NaN", True)
        True
        >>> is_value_nan(float("nan"))
        True
    """
    if loose:
        try:
            return math.isnan(float(value))  # type: ignore[arg-type]
        except (ValueError, TypeError):
            return False
    return isinstance(value, float) and math.isnan(value)
